components/camel-elasticsearch5-rest/src/main/java/org/apache/camel/component/elasticsearch5/converter/ElasticsearchActionRequestConverter.java

```diff
@@ -0,0 +1,211 @@
+        return new GetRequest(exchange.getIn().getHeader(
+            ElasticsearchConstants.PARAM_INDEX_NAME, String.class))
+            .type(exchange.getIn().getHeader(
+                ElasticsearchConstants.PARAM_INDEX_TYPE,
+                String.class)).id(id);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Converter
+    public static MultiSearchRequest toMultiSearchRequest(Object document, Exchange exchange) {
+        List<SearchRequest> items = (List<SearchRequest>) document;
+        MultiSearchRequest multiSearchRequest = new MultiSearchRequest();
+        Iterator<SearchRequest> it = items.iterator();
+        while (it.hasNext()) {
+            SearchRequest item = it.next();
+            multiSearchRequest.add(item);
+        }
+        return multiSearchRequest;
+    }
+
+    @Converter
+    public static DeleteRequest toDeleteRequest(String id, Exchange exchange) {
+        return new DeleteRequest()
+            .index(exchange.getIn().getHeader(
+                ElasticsearchConstants.PARAM_INDEX_NAME,
+                String.class))
+            .type(exchange.getIn().getHeader(
+                ElasticsearchConstants.PARAM_INDEX_TYPE,
+                String.class)).id(id);
+    }
+
+    @Converter
+    public static SearchRequest toSearchRequest(Object queryObject, Exchange exchange) {
+        SearchRequest searchRequest = new SearchRequest(exchange.getIn()
+            .getHeader(ElasticsearchConstants.PARAM_INDEX_NAME, String.class))
+            .types(exchange.getIn().getHeader(ElasticsearchConstants.PARAM_INDEX_TYPE, String.class));
+
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        String queryText = null;
+
+        if (queryObject instanceof Map<?, ?>) {
+            Map<String, Object> mapQuery = (Map<String, Object>)queryObject;
+            // Remove 'query' prefix from the query object for backward compatibility
+            if (mapQuery.containsKey(ElasticsearchConstants.ES_QUERY_DSL_PREFIX)) {
+                mapQuery = (Map<String, Object>)mapQuery.get(ElasticsearchConstants.ES_QUERY_DSL_PREFIX);
+            }
+            try {
+                XContentBuilder contentBuilder = XContentFactory.contentBuilder(XContentType.JSON);
+                queryText = contentBuilder.map(mapQuery).string();
+            } catch (IOException e) {
+                LOG.error(e.getMessage());
+            }
+        } else if (queryObject instanceof String) {
+            queryText = (String)queryObject;
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                JsonNode jsonTextObject = mapper.readValue(queryText, JsonNode.class);
+                JsonNode parentJsonNode = jsonTextObject.get(ElasticsearchConstants.ES_QUERY_DSL_PREFIX);
+                if (parentJsonNode != null) {
+                    queryText = parentJsonNode.toString();
+                }
+            } catch (IOException e) {
+                LOG.error(e.getMessage());
```

Comment:
We should not just ignore exceptions like that. Always fail by letting the exception be thrown